### PR TITLE
Grab bag of CircleCI / deployment fixes

### PR DESCRIPTION
**Pull request by ulope**
This includes a few fixes / improvements for CircleCI and deployment:

- Add check for consistent CircleCI workflows
  Prevents the `raiden-default` and `nightly` workflows from diverging
- Allow skipping tests via commit message tag `[skip tests]`
  Other than the builtin `[skip ci]` tag this prevents the tests from running but will still report a valid commit status back to GitHub
- Fix missing job dependency for CircleCI macOS deployment
  The nightly macOS deploy job was missing a dependency
- Prevent PyInstaller from creating broken binaries 
  When `pdbpp` is installed PyInstaller creates broken binaries due to import path hacking performed by pdbpp

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -78,13 +85,73 @@ commands:
...
+          command: python setup.py egg_info
+
+
+      - restore_cache:
+          key: system-deps-geth-{{ checksum "/tmp/geth-version" }}-solc-{{ checksum "/tmp/solc-version" }}
+
+      - run:
+          name: Installing system dependencies
+          command: |
+            GETH_PATH="${HOME}/.local/bin/geth-${OS_NAME}-${GETH_VERSION}"
```
Comment (LefterisJP) [inline .circleci/config.yml]: Wouldn't it be cleaner and more maintainale to have this script into a separate bash file?

File: raiden.spec [hunk 2]
```diff
@@ -60,6 +60,13 @@ def Entrypoint(dist, group, name, scripts=None, pathex=None, hiddenimports=None,
     )
 
 
+if hasattr(pdb, 'pdb'):
+    # pdbpp moves the stdlib pdb to the `pdb` attribute of it's own patched pdb module
+    raise RuntimeError(
+        'pdbpp is installed with causes broken PyInstaller builds. Please uninstall it.',
```
Comment (LefterisJP) [inline raiden.spec]: /s/with/which

File: tools/check_circleci_config.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    return click.style(string, fg='green')
+
+
+def _yellow(string):
+    return click.style(string, fg='yellow')
+
+
+def _check_workflows_align(config):
+    """
+    Ensure that the common shared jobs in the `raiden-default` and `nightly` workflows are
```
Comment (LefterisJP) [inline tools/check_circleci_config.py]: nice idea. Curious question. Is there any occasion where they fell out of sync and prompted you to introduce this script as a check?

File: .circleci/conditional_skip.sh [hunk 4]
```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+
+SKIP_TAG="\[skip tests\]"
```
Comment (LefterisJP) [inline .circleci/conditional_skip.sh]: Can't we also have the same behaviour with `[skip ci]`?

Comment (ulope) [reply .circleci/conditional_skip.sh]: No, both `[skip ci]` and `[ci skip]` are hardcoded and prevent the entire ci from running.

File: .circleci/config.yml (hunk 1 above)
Comment (LefterisJP) [reply .circleci/config.yml]: Would be nice

File: tools/check_circleci_config.py (hunk 3 above)
Comment (LefterisJP) [reply tools/check_circleci_config.py]: :+1:

File: .circleci/conditional_skip.sh (hunk 4 above)
Comment (LefterisJP) [reply .circleci/conditional_skip.sh]: Ah damn that sucks. So we should avoid [skip ci] from here and on then.

**Pull request merged**